rest/config.go

```diff
@@ -695,6 +696,13 @@ func (dbConfig *DbConfig) validateVersion(isEnterpriseEdition bool) error {
 		}
 	}
 
+	// Ensure that, if password auth is disabled, there's another way to authenticate
+	if dbConfig.DisablePasswordAuth {
+		if dbConfig.OIDCConfig == nil && (dbConfig.Guest == nil || base.BoolDefault(dbConfig.Guest.Disabled, true)) {
+			multiError = multiError.Append(fmt.Errorf("password authentication is disabled, but there is no alternative auth configuration (OIDC or guest access)"))
+		}
+	}
```

Comment:
There is one more authentication mechanism customers may use outside of OIDC or guest which means we can't do this check.
Admin-created sessions for users (i.e. `:4985/db/_session` -> `createUserSession()`). This might be hooked up to some external non-OIDC auth service that doles out sessions for users.